modules/substrate/src/lib.rs

```diff
@@ -361,9 +403,9 @@ impl<T: Trait> BridgeStorage for PalletStorage<T> {
 		CurrentAuthoritySet::put(new_set)
 	}
 
-	fn enact_authority_set(&mut self) -> Result<(), ()> {
-		if <NextScheduledChange<T>>::exists() {
-			let new_set = <NextScheduledChange<T>>::take()
+	fn enact_authority_set(&mut self, signal_hash: BridgedBlockHash<T>) -> Result<(), ()> {
+		if <NextScheduledChange<T>>::contains_key(signal_hash) {
+			let new_set = <NextScheduledChange<T>>::take(signal_hash)
```

Comment:
The reason I did this is because `take()` requires decoding the storage value, whereas `contains_key()` doesn't. This would help in the case where we don't have the key.

In `verifier` we only call this method in the happy-path, so maybe this optimization isn't really worth it in practice.

Reply:
But if you `take` and there is `None`, you don't have to decode anything (`contains_key==false`), if `contains_key==true` you decode the value anyway, so there is actaully one storage read (existence check) that you save with just `take`.

Reply:
I just took a look at the storage code and you're right. When you `take()` it doesn't decode unless there was a value at the given key